src/pages/index.js

```diff
@@ -74,43 +150,136 @@ previewModalCloseBtn.addEventListener("click", () => {
-    previewModalCaptionElement.textContent = data.name;
+  // Add click event listener for like button
+  cardLikeButton.addEventListener("click", () => {
+    handleLikeClick(cardData._id, cardLikeButton);
   });
-
   return cardElement;
 }
 
+// Open Avatar Modal
+profileAvatarBtn.addEventListener("click", () => {
+  resetValidation(avatarForm, settings);
+  openModal(avatarModal);
+});
+
+// Close Avatar Modal
+avatarModalCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+
+// Avatar Form Submission
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
 function openModal(modal) {
   modal.classList.add("modal_opened");
   document.addEventListener("keydown", handleEscapeKey);
 }
 
+// Avatar
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarForm.querySelector(".modal__submit-btn");
+  const avatarImg = document.querySelector(".profile__avatar");
+
+  // Set loading state
+  submitButton.disabled = true;
+  submitButton.classList.add("modal__submit-btn_loading");
+  submitButton.textContent = "Saving...";
+
+  api
+    .updateAvatar({ avatar: avatarUrlInput.value })
+    .then((userData) => {
+      avatarImg.src = "";
+      avatarImg.src = `${userData.avatar}?t=${Date.now()}`;
+
+      // Reset form and close modal
+      avatarForm.reset();
+      closeModal(avatarModal);
+
+      // Reset validation state
+      resetValidation(avatarForm, settings);
+
+      // Reset button state
+      submitButton.classList.remove("modal__submit-btn_loading");
+      submitButton.textContent = "Save";
+      disableButton(submitButton, settings);
+    })
+    .catch((err) => {
+      console.error("Avatar update failed:", err);
+      alert("Failed to update avatar. Please try again.");
+
+      // Reset button state but keep current validation
+      submitButton.classList.remove("modal__submit-btn_loading");
+      submitButton.textContent = "Save";
+      submitButton.disabled = false;
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = cardForm.querySelector(".modal__submit-btn");
+
+  // Set loading state
+  submitButton.disabled = true;
+  submitButton.classList.add("modal__submit-btn_loading");
+  submitButton.textContent = "Creating...";
+
+  api
+    .addCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+
+      // Reset form and close modal
+      cardForm.reset();
+      closeModal(cardModal);
+
+      // Reset validation state
+      resetValidation(cardForm, settings);
+    })
+    .catch((err) => {
+      console.error("Failed to add card:", err);
+    })
+    .finally(() => {
+      // Reset button state
+      submitButton.classList.remove("modal__submit-btn_loading");
+      submitButton.textContent = "Create";
+      disableButton(submitButton, settings);
```

Comment:
` disableButton(submitButton, settings);`

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.